File: CS50x/pset6/cash/cash.py
from cs50 import get_float
import numpy as np

# variables for program
coins = 0
money = 0.00
centAdded = 0.00

# array of cent amount to add
# np.arange(0., 1,  )
cents = np.array([0.25, 0.10, 0.05, 0.01], dtype=np.float)

# entering do while lood 
while True:
    # get input from user 
    money = get_float("Change owed: ")
    
    # iterate to next loop
    if money < 0.0:
        continue

    # calculate coins 
    for i in range(4):
        while True:
            # if less then add if more
            # subtract and break
            # if equal break
            if float('%.2f' % centAdded) < float('%.2f' % money):
                centAdded += cents[i]
                coins += 1
            elif float('%.2f' % centAdded) > float('%.2f' % money):
                centAdded -= cents[i]
                coins -= 1
                break
            else:  # if equal 
                break

    # break out of the do while            
    if money > 0.0:
        break
    
print(coins)

File: CS50x/pset1/cash/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    /*
     *coins is the return int
     *centadded is for the evaluation
     */
    int coins = 0;
    float money = 0;
    float centAdded = 0;

    /*cents are a const float to use for adding
     *this way i don't have to worry about float imprecision
     */
    const float cents[4] = 
    {
        0.25f, 
        0.10f, 
        0.05f,
        0.01f
    };

    do
    {
        money = get_float("how much owd: ");

        //makes sure to continue loop 
        //to the next iteration if negative value
        if (money < 0.0f)
        {
            continue;
        }

        for (int i = 0; i < 4; ++i)
        {
            /*this float is to add coins*/
            for (; ; ++coins)
            {
                //if less add if more subtract and break
                //if equal break
                if (centAdded < money)
                {
                    centAdded += cents[i];
                }
                else if (centAdded > money)
                {
                    centAdded -= cents[i];
                    coins --;
                    break;
                }
                else //if centAdded equal to money
                {
                    break;
                }

            }
        }
    }
    while (money < 0.00f);

    //output how many coins
    printf("%i\n", coins);
}
